### ISSUE TITLE
Preview deployments for Workers

### ISSUE THREAD

**Issue by gentlementlegen** [ubiquibot/plugins-wishlist#36]
This is not specific plugin but a nice-to-have features for plugin Workers.

Currently when developing it's quite easy to switch to a different Action branch for testing a new feature, by specifying the `@branch` within the configuration.

However, workers do not benefit from this and always have to be deployed from the main branch to be tested. This makes development more tedious than it has to be, even worse when testing other people pull-requests.

What would be nice is to get a preview URL base on the latest commit of the branch the user is working on.

It would also be important to cleanup the preview deployments as they would pollute Cloudflare heavily.

A naming pattern could be:
`<plugin-name>.<username>.<branch>.<hash>` to find them easily and know what they refer to. We could also have the deployed url displayed in the Action results to be found easily, and without the need to access Github token because we do not publish in the pull-request thread.

**Comment by 0x4007** [comment]
> What would be nice is to get a preview URL base on the latest commit of the branch the user is working on.

Your technical specification does not offer clarity into how you plan to implement this. Otherwise I would suggest just doing local development (can you use localhost in the bot config?) and then finally doing a deploy?

**Comment by gentlementlegen** [comment]
I think an Action can handle it all, we already have the Cloudflare deployment set up. The idea is that instead of having every time to clone the user's branch, set it up locally etc. we could just try it through the URL. That would also allow testing with multiple versions without having them merged on the main branch all the time (every time I test someone else's PR I need to force push my main branch on my org). And for safety we could easily test any deployment with the `sandbox` repo as well.

**Comment by 0x4007** [comment]
We could just ask that contributors allow us to test on their own bot on their own org. No extra work needed.

**Comment by gentlementlegen** [comment]
Yes but still we cannot handle multi branch on our own repos, so we always need to deploy the main branch, which is what I am trying to solve here 😄

**Comment by gentlementlegen** [comment]
If this gets merged, we might want to have this running on our Worker plugins. Could be helpful to get https://github.com/ubiquity/ts-template/pull/56 to help with this.
